OpenTrons/opentrons-api
# feat(odd): add run failed modal for running protocol

# Overview
Add run failed modal. When runStatus is failed, the app shows the modal.

The current modal is different from the design since we need a new error handling system to display the information in the Figma. This PR is using the first of errors from `run/${runId}`. (Also error codes aren't defined yet.)
When the system is ready, we will be able to align this modal with the design.

I comment out the event for `stopRun` and I will follow up this with Shlok, Jesse, and Anthony.

![Screenshot 2023-04-12 at 5 46 35 PM](https://user-images.githubusercontent.com/474225/231600071-2da26f44-e1a1-4754-a50f-05734ce31090.png)


[design]
https://www.figma.com/file/AoTLAYuWawlaWItB1umOjr/Release%3A-Opentrons-Flex-Touchscreen-Designs?node-id=8860-128824&t=I3cn6xOFvLdgrX4y-0

close RCORE-563


# Test Plan
- Send a protocol that an OT-3 or dev kit is missing something.
- Tap All Protocols
- Tap the protocol you sent
- Tap run
Wait for a little bit and check the modal is showed up



# Changelog
- add a new component and test for this modal
- update Modal component for isError


# Review requests



# Risk assessment
low
